airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StreamResetPersistence.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.persistence;
+
+import static org.jooq.impl.DSL.noCondition;
+
+import io.airbyte.config.StreamKey;
+import io.airbyte.config.StreamResetRecord;
+import io.airbyte.db.Database;
+import io.airbyte.db.ExceptionWrappingDatabase;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.jooq.Condition;
+import org.jooq.Record;
+import org.jooq.RecordMapper;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+
+public class StreamResetPersistence {
+
+  private static final String ID_COL = "id";
+  private static final String STREAM_NAME_COL = "stream_name";
+  private static final String STREAM_NAMESPACE_COL = "stream_namespace";
+  private static final String CONNECTION_ID_COL = "connection_id";
+  private static final String CREATED_AT_COL = "created_at";
+  private static final String UPDATED_AT_COL = "updated_at";
+  private static final Table<Record> DSL_TABLE_STREAM_RESET = DSL.table("stream_reset");
+
+  private final ExceptionWrappingDatabase database;
+
+  public StreamResetPersistence(final Database database) {
+    this.database = new ExceptionWrappingDatabase(database);
+  }
+
+  public List<StreamKey> getStreamResets(final UUID connectionId) throws IOException {
+    return database.query(ctx -> ctx.select(DSL.asterisk())
+        .from(DSL_TABLE_STREAM_RESET))
+        .where(DSL.field(CONNECTION_ID_COL).eq(connectionId))
```

Comment:
I think that we should have an index on `connection_id` instead of `(connection_id, stream_name, stream_namespace)` based on this request. And also for the other request in the PR. This should be in another PR.

Reply:
Oh, I assumed that a (connection_id, stream_name, stream_namespace) index would also mean there is an index on connection_id since that is the first column in the index. I've really only worked with MySQL in the past though and not postgres, so let me check.

Reply:
I think this is fine - queries on just connection_id should take advantage of the index: https://www.postgresql.org/docs/current/indexes-multicolumn.html